examples/pattern-matcher/define.scm

```diff
@@ -0,0 +1,63 @@
+;
+; define.scm
+;
+; Demonstrate the use of DefineLink to give names to things.
+;
+; XXX caution: under construction, unstable, known buggy.
+;
+(use-modules (opencog))
+(use-modules (opencog query))
+
+(load-from-path "utilities.scm")
+
+;; Some data to populate the atomspace.
+(InheritanceLink
+	(ConceptNode "battery")
+	(ConceptNode "electrical device"))
+
+(InheritanceLink
+	(ConceptNode "transistor")
+	(ConceptNode "electrical device"))
+
+(EvaluationLink
+	(PredicateNode "PartOf")
+	(ListLink
+		(ConceptNode "battery")
+		(VariableNode "car")))
+
+(EvaluationLink
+	(PredicateNode "PartOf")
+	(ListLink
+		(ConceptNode "transistor")
+		(VariableNode "phone")))
+
+(EvaluationLink
+	(PredicateNode "PartOf")
+	(ListLink
+		(ConceptNode "windsheild")
+		(VariableNode "car")))
+
+;; Define the concept of electrical parts of things.
+(define elect-parts
+	(DefineLink
+		(DefinedPredicateNode "Electrical Part Of")
+		(PresentLink
+			(InheritanceLink
+				(VariableNode "$x")
+				(ConceptNode "electrical device"))
+			(EvaluationLink
+				(PredicateNode "PartOf")
+				(ListLink
+					(VariableNode "$x")
+					(VariableNode "$y"))))))
+
+;; Define a pattern to find the electrical parts of things
+;; At this time, an explicit variable declaration has to be done;
+;; this needs to be fixed...
+(define get-elect
+	(GetLink 
+		(VariableList (VariableNode "$x") (VariableNode "$y"))
+		(DefinedPredicateNode "Electrical Part Of")))
```

Comment:
Yes, that would be better. This, however, has been one of the issues confusing me. It requires a more sophisticated "pasting" rule, when assembling a pattern out of define links.  The pasting rule seems to become: "if a lambda link is found, then paste the body of the lambda, not the lambda itself".  It's ... seems kind of strange at first, but it seems to be the right thing to o, I guess.